src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1163,6 +1163,11 @@
     "create index fact_values_value_float_idx on fact_values(value_float)"
     "create index fact_values_value_integer_idx on fact_values(value_integer)"))
 
+(defn add-corrective-change-index
+  []
+  (jdbc/do-commands
+    "CREATE INDEX resource_events_status_for_corrective_change_idx ON resource_events (status) WHERE corrective_change;"))
```

Comment:
The purpose of this index is to speed up queries with `corrective_change = true`. Just a small percentage of events should be corrective so the partial index seems to be more appropriate here. When PE queries `resource_events` by `corrective_change` it always specifies `status` too (e.g. `[:and [:= :status "success"] [:= :corrective_change true]]` that is why the index is for `status`. IMHO, a multicolumn index for `status` and `corrective_change` is not needed since `corrective_change` is always true in case of this partial index.